### ISSUE TITLE
[Bug]: standalone panic with error `fatal error: concurrent map read and map write` during test

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#33060]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20240514-96489b81-amd64
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka): rocksmq   
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2024/05/14 09:36:55.615 +00:00] [INFO] [delegator/delegator_data.go:346] ["add growing segments to delegator"] [collectionID=449755009669567726] [channel=by-dev-rootcoord-dml_0_449755009669567726v0] [replicaID=449755009822490661] [segmentIDs="[449755009669568110]"]
[2024/05/14 09:36:55.615 +00:00] [DEBUG] [delegator/delegator_data.go:160] ["insert into growing segment"] [collectionID=449755009669567726] [channel=by-dev-rootcoord-dml_0_449755009669567726v0] [replicaID=449755009822490661] [collectionID=449755009669567726] [segmentID=449755009669568110] [rowCount=3000] [maxTimestamp=449755064647024641]
[2024/05/14 09:36:55.620 +00:00] [INFO] [datacoord/meta.go:1256] ["meta update: alter in memory meta after compaction - complete"] [planID=449755009669567263] [type=MixCompaction] [channel=by-dev-rootcoord-dml_11_449755009668164387v0] [partitionID=449755009668164388] ["compactTo segmentID"=449755009667766371] ["compactTo segment numRows"=8998] ["compactFrom segments(to be updated as dropped)"="[449755009669366965,449755009669366318,449755009669165409,449755009668765118,449755009669366015]"]
[2024/05/14 09:36:55.620 +00:00] [INFO] [datacoord/compaction.go:507] ["handleCompactionResult: syncing segments with node"] [planID=449755009669567263] [nodeID=1]
[2024/05/14 09:36:55.620 +00:00] [INFO] [datanode/services.go:296] ["DataNode receives SyncSegments"] [planID=449755009669567263] [nodeID=1] ["target segmentID"=449755009667766371] ["compacted from"="[449755009669366965,449755009669366318,449755009669165409,449755009668765118,449755009669366015]"] [numOfRows=8998] [channelName=by-dev-rootcoord-dml_11_449755009668164387v0]
[2024/05/14 09:36:55.620 +00:00] [INFO] [datanode/data_sync_service.go:255] ["begin to init pk bloom filter"] [segmentID=449755009667766371] [statsBinLogsLen=1]
fatal error: concurrent map read and map write
[2024/05/14 09:36:55.621 +00:00] [INFO] [datacoord/meta.go:168] ["meta update: add collection"] [collectionID=449755009669568117]

goroutine 75480 [running]:
runtime.fatal({0x558e7b3?, 0xc0072cc9b0?})
	/usr/local/go/src/runtime/panic.go:1066 +0x5d fp=0xc006689ed0 sp=0xc006689ea0 pc=0x1c2445d
runtime.mapaccess2_fast64(0x4e84b80, 0xc001f9fcb0, 0x63dd9cc63392b23)
	/usr/local/go/src/runtime/map_fast64.go:62 +0x57 fp=0xc006689ef0 sp=0xc006689ed0 pc=0x1bfb777
github.com/milvus-io/milvus/internal/datacoord.(*meta).GetCollectionIndexFilesSize(0xc000309490)
	/go/src/github.com/milvus-io/milvus/internal/datacoord/meta.go:326 +0xce fp=0xc00668a030 sp=0xc006689ef0 pc=0x41d7a8e
github.com/milvus-io/milvus/internal/datacoord.(*Server).getQuotaMetrics(0xc00002ac80)
	/go/src/github.com/milvus-io/milvus/internal/datacoord/metrics_info.go:42 +0x4f fp=0xc00668a068 sp=0xc00668a030 pc=0x41e740f
github.com/milvus-io/milvus/internal/datacoord.(*Server).getDataCoordMetrics(_, {_, _})
	/go/src/github.com/milvus-io/milvus/internal/datacoord/metrics_info.go:168 +0x32c fp=0xc00668a368 sp=0xc00668a068 pc=0x41e890c
github.com/milvus-io/milvus/internal/datacoord.(*Server).getSystemInfoMetrics(0xc00002ac80, {0x5b53af8, 0xc006fc2150}, 0x1bff850?)
	/go/src/github.com/milvus-io/milvus/internal/datacoord/metrics_info.go:96 +0x8f fp=0xc00668ae60 sp=0xc00668a368 pc=0x41e7bef
github.com/milvus-io/milvus/internal/datacoord.(*Server).GetMetrics(0xc00002ac80, {0x5b53af8, 0xc006fc2150}, 0xc00cba4c40)
	/go/src/github.com/milvus-io/milvus/internal/datacoord/services.go:1041 +0x60c fp=0xc00668b3e0 sp=0xc00668ae60 pc=0x4211e0c
github.com/milvus-io/milvus/internal/distributed/datacoord.(*Server).GetMetrics(0xc018818de0?, {0x5b53af8?, 0xc006fc2150?}, 0xc00360d788?)
	/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/service.go:355 +0x2f fp=0xc00668b410 sp=0xc00668b3e0 pc=0x423a54f
github.com/milvus-io/milvus/internal/proto/datapb._DataCoord_GetMetrics_Handler.func1({0x5b53af8, 0xc006fc2150}, {0x5257da0?, 0xc00cba4c40})
	/go/src/github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:7547 +0x7b fp=0xc00668b450 sp=0xc00668b410 pc=0x2b0c37b
github.com/milvus-io/milvus/pkg/util/interceptor.ServerIDValidationUnaryServerInterceptor.func1({0x5b53af8, 0xc006fc2150}, {0x5257da0, 0xc00cba4c40}, 0x7?, 0xc003fad6e0)
	/go/src/github.com/milvus-io/milvus/pkg/util/interceptor/server_id_interceptor.go:54 +0x102 fp=0xc00668b4a0 sp=0xc00668b450 pc=0x3eef7c2
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x5b53af8?, 0xc006fc2150?}, {0x5257da0?, 0xc00cba4c40?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a fp=0xc00668b4e0 sp=0xc00668b4a0 pc=0x27e85ba
github.com/milvus-io/milvus/pkg/util/interceptor.ClusterValidationUnaryServerInterceptor.func1({0x5b53af8, 0xc006fc2150}, {0x5257da0, 0xc00cba4c40}, 0x0?, 0xc004736960)
	/go/src/github.com/milvus-io/milvus/pkg/util/interceptor/cluster_interceptor.go:48 +0xf3 fp=0xc00668b540 sp=0xc00668b4e0 pc=0x3ef0533
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x5b53af8?, 0xc006fc2150?}, {0x5257da0?, 0xc00cba4c40?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a fp=0xc00668b580 sp=0xc00668b540 pc=0x27e85ba
github.com/milvus-io/milvus/pkg/util/logutil.UnaryTraceLoggerInterceptor({0x5b53af8?, 0xc006fc2150?}, {0x5257da0, 0xc00cba4c40}, 0xc004ebb5e0?, 0xc0047369a0)
	/go/src/github.com/milvus-io/milvus/pkg/util/logutil/grpc_interceptor.go:23 +0x49 fp=0xc00668b5b0 sp=0xc00668b580 pc=0x3b3d489
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x5b53af8?, 0xc006fc2150?}, {0x5257da0?, 0xc00cba4c40?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a fp=0xc00668b5f0 sp=0xc00668b5b0 pc=0x27e85ba
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1({0x5b53af8, 0xc006fc2090}, {0x5257da0, 0xc00cba4c40}, 0xc004736940, 0xc0047369c0)
	/go/pkg/mod/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.38.0/interceptor.go:342 +0x528 fp=0xc00668b960 sp=0xc00668b5f0 pc=0x24e61e8
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x5b53af8?, 0xc006fc2090?}, {0x5257da0?, 0xc00cba4c40?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a fp=0xc00668b9a0 sp=0xc00668b960 pc=0x27e85ba
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x5b53af8, 0xc006fc2090}, {0x5257da0, 0xc00cba4c40}, 0xc01f8d6a40?, 0x4edc6c0?)
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xbe fp=0xc00668b9f8 sp=0xc00668b9a0 pc=0x27e845e
github.com/milvus-io/milvus/internal/proto/datapb._DataCoord_GetMetrics_Handler({0x545a380?, 0xc0016189c0}, {0x5b53af8, 0xc006fc2090}, 0xc003fcc8c0, 0xc0013c4390)
	/go/src/github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:7549 +0x138 fp=0xc00668ba50 sp=0xc00668b9f8 pc=0x2b0c238
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0017c2000, {0x5b75aa0, 0xc0021d4820}, 0xc002ab67e0, 0xc0013c45a0, 0x7ad6b50, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.57.1/server.go:1358 +0xe23 fp=0xc00668be30 sp=0xc00668ba50 pc=0x235fc83
google.golang.org/grpc.(*Server).handleStream(0xc0017c2000, {0x5b75aa0, 0xc0021d4820}, 0xc002ab67e0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.57.1/server.go:1735 +0xa36 fp=0xc00668bf50 sp=0xc00668be30 pc=0x2364dd6
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	/go/pkg/mod/google.golang.org/grpc@v1.57.1/server.go:970 +0xca fp=0xc00668bfe0 sp=0xc00668bf50 pc=0x235d5aa
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1598 +0x1 fp=0xc00668bfe8 sp=0xc00668bfe0 pc=0x1c5dec1
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.57.1/server.go:981 +0x15c
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/deploy_test_cron/detail/deploy_test_cron/2285/pipeline
log:
[artifacts-rocksmq-standalone-reinstall-2285-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/15315681/artifacts-rocksmq-standalone-reinstall-2285-server-logs.tar.gz)

**Comment by congqixia** [comment]
shall be fixed in #33029
/assign @zhuwenxing 
could you please verify?

**Comment by zhuwenxing** [comment]
verified and fixed in `master-20240515-d4837307-amd64`
